src/core/heila-core.js

```diff
@@ -0,0 +1,311 @@
+    .select('users.*')
+    .where({ id: userId })
+    .then(userRows => {
+      if (_.isEmpty(userRows)) {
+        return [];
+      }
+      const heilaIds = userRows.map(user => {
+        return user.uuid;
+      })
+      return knex('heilas')
+        .select('heilas.*')
+        .whereIn('uuid', heilaIds)
+        .then(heilaRows => {
+          if (_.isEmpty(heilaRows)) {
+            return [];
+          }
+          return _heilaRowsToObjectList(_mergeUserHeilaRows(userRows, heilaRows))[0];
+        })
+    })
+
+}
+
+function _mergeUserHeilaRows(userRows, heilaRows) {
+  // mergetään user-profiilit ja heila-profiilit
+  // lopputuloksena palautetaan lista, jossa
+  // sekä userin tiedot että heilan tiedot samassa
+  // objektissa
+  userRows.forEach(user => {
+    for (let i = 0; i < heilaRows.length; i += 1) {
+      if (user.uuid === heilaRows[i].uuid) {
+        user = Object.assign(user, heilaRows[i]);
+        break;
+      }
+    }
+  })
+  return userRows;
+}
+
+function _heilaRowsToObjectList(userList) {
+
+  // userList sisältää _VAIN_ heila: true -tyyppisiä profiileja
+
+  const heilaList = userList
+    .map(user => {
+      return {
+        id: user.id,
+        name: user.name,
+        team_id: user.team_id,
+        image_url: user.image_path ? prefixImageWithGCS(user.image_path) : null,
+        bio_text: user.bio_text,
+        bio_looking_for_type_id: user.bio_looking_for_type_id,
+        class_year: user.class_year,
+      }
+    })
+  return heilaList;
+}
+
+function _makeHeilaDbRow(heila) {
+  const dbRow = {
+    'userId': heila.userId,
+    'uuid': heila.uuid,
+    'bio_text': heila.bio_text,
+    'bio_looking_for_type_id': heila.bio_looking_for_type_id,
+    'pushToken': heila.pushToken,
+    'class_year': heila.class_year,
+  };
+  return dbRow;
+}
+
+// jscs:disable requireCamelCaseOrUpperCaseIdentifiers
+function _heilaRowToObject(row) {
+  // jos kuvaa ei ole uploadattu, palautetaan tyhjä merkkijono
+  let url = null;
+  if (row.image_path) {
+    url = `${GCS_CONFIG.baseUrl}/${GCS_CONFIG.bucketName}/${row.image_path}`;
+  }
+
+  return {
+    id: row.userId,
+    uuid: row.uuid,
+    // tässä asetetaan image_urli, jolla sen voi hakea verkosta ja näyttää
+    // urlia ei oikeasti ole laitettu kantaan, siellä on vain image_path
+    image_url: url,
+    bio_text: row.bio_text,
+    bio_looking_for_type_id: row.bio_looking_for_type_id,
+    class_year: row.class_year,
+  };
+}
+
+function getHeilaTypes() {
+  return knex('heila_types')
+    .select('heila_types.*')
+    .then(rows => {
+      return rows;
+    });
+};
+
+function deleteHeila(uuid) {
+  // first drops the row from heilas table
+  // then toggles the true/false of users table
+  // for this uuid
+  // this IS a side-effect if you think about it
+  // from the users table's perspective, yes.
+  return knex('heilas')
+    .where({ 'uuid': uuid })
+    .del()
+    .then(res => {
+      // TODO: could this Promise Hell be somehow improved?!
```

Comment:
It could be, by flattening the promise chain 😎 Or using async/await, since it's available in > Node 8.x